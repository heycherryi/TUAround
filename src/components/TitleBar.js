import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { faRoute, faInfoCircle, faBicycle } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import logo from '../resources/logo.svg';

function TitleBar({ hide }) {
  if (hide) {
    return;
  }
  return (
    <div>
      <div className="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 navbar-dark bg-dark border-bottom shadow-sm">
          <Link to="/" className="text-light h3 mb-0"><img src={logo} width="30" height="30" className="d-inline-block align-top me-2" alt="Bus" /> WeGo Transit Map</Link>
        <div className="h5 my-0 me-md-auto">
        </div>
        <nav className="my-2 my-md-0 mr-md-3">
          <Link className="p-2 text-light" to="/about"><FontAwesomeIcon icon={faInfoCircle} fixedWidth={true} /> About</Link>
          <Link className="p-2 text-light" to="/routes"><FontAwesomeIcon icon={faRoute} fixedWidth={true} /> Routes</Link>
          <Link className="p-2 text-light" to="/bcycle"><FontAwesomeIcon icon={faBicycle} fixedWidth={true} /> BCycle</Link>
        </nav>
      </div>
    </div>
  );
}

TitleBar.propTypes = {
  hide: PropTypes.bool,
};

TitleBar.defaultProps = {
  hide: false,
};

export default TitleBar;
