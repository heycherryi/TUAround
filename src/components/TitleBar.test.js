/* globals test, expect */

import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import TitleBar from './TitleBar';

test('renders TitleBar', () => {
  const { container } = render(
    <Router>
      <TitleBar />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});
