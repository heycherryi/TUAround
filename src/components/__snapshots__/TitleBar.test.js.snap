// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TitleBar 1`] = `
<div>
  <div>
    <div
      class="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 navbar-dark bg-dark border-bottom shadow-sm"
    >
      <a
        class="text-light h3 mb-0"
        href="/"
      >
        <img
          alt="Bus"
          class="d-inline-block align-top me-2"
          height="30"
          src="logo.svg"
          width="30"
        />
         WeGo Transit Map
      </a>
      <div
        class="h5 my-0 me-md-auto"
      />
      <nav
        class="my-2 my-md-0 mr-md-3"
      >
        <a
          class="p-2 text-light"
          href="/about"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-info fa-fw "
            data-icon="circle-info"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
              fill="currentColor"
            />
          </svg>
           About
        </a>
        <a
          class="p-2 text-light"
          href="/routes"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-route fa-fw "
            data-icon="route"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96c0 50.2-59.1 125.1-84.6 155c-3.8 4.4-9.4 6.1-14.5 5H320c-17.7 0-32 14.3-32 32s14.3 32 32 32h96c53 0 96 43 96 96s-43 96-96 96H139.6c8.7-9.9 19.3-22.6 30-36.8c6.3-8.4 12.8-17.6 19-27.2H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320c-53 0-96-43-96-96s43-96 96-96h39.8c-21-31.5-39.8-67.7-39.8-96c0-53 43-96 96-96s96 43 96 96zM117.1 489.1c-3.8 4.3-7.2 8.1-10.1 11.3l-1.8 2-.2-.2c-6 4.6-14.6 4-20-1.8C59.8 473 0 402.5 0 352c0-53 43-96 96-96s96 43 96 96c0 30-21.1 67-43.5 97.9c-10.7 14.7-21.7 28-30.8 38.5l-.6 .7zM128 352a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zM416 128a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"
              fill="currentColor"
            />
          </svg>
           Routes
        </a>
        <a
          class="p-2 text-light"
          href="/bcycle"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bicycle fa-fw "
            data-icon="bicycle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M312 32c-13.3 0-24 10.7-24 24s10.7 24 24 24h25.7l34.6 64H222.9l-27.4-38C191 99.7 183.7 96 176 96H120c-13.3 0-24 10.7-24 24s10.7 24 24 24h43.7l22.1 30.7-26.6 53.1c-10-2.5-20.5-3.8-31.2-3.8C57.3 224 0 281.3 0 352s57.3 128 128 128c65.3 0 119.1-48.9 127-112h49c8.5 0 16.3-4.5 20.7-11.8l84.8-143.5 21.7 40.1C402.4 276.3 384 312 384 352c0 70.7 57.3 128 128 128s128-57.3 128-128s-57.3-128-128-128c-13.5 0-26.5 2.1-38.7 6L375.4 48.8C369.8 38.4 359 32 347.2 32H312zM458.6 303.7l32.3 59.7c6.3 11.7 20.9 16 32.5 9.7s16-20.9 9.7-32.5l-32.3-59.7c3.6-.6 7.4-.9 11.2-.9c39.8 0 72 32.2 72 72s-32.2 72-72 72s-72-32.2-72-72c0-18.6 7-35.5 18.6-48.3zM133.2 368h65c-7.3 32.1-36 56-70.2 56c-39.8 0-72-32.2-72-72s32.2-72 72-72c1.7 0 3.4 .1 5.1 .2l-24.2 48.5c-9 18.1 4.1 39.4 24.3 39.4zm33.7-48l50.7-101.3 72.9 101.2-.1 .1H166.8zm90.6-128H365.9L317 274.8 257.4 192z"
              fill="currentColor"
            />
          </svg>
           BCycle
        </a>
      </nav>
    </div>
  </div>
</div>
`;
